internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -418,7 +418,14 @@ func (r *Reconciler) extensionConfigToClusterClass(ctx context.Context, o client
 		}
 		for _, patch := range clusterClass.Spec.Patches {
 			if patch.External != nil && patch.External.DiscoverVariablesExtension != nil {
-				res = append(res, ctrl.Request{NamespacedName: client.ObjectKey{Namespace: clusterClass.Namespace, Name: clusterClass.Name}})
+				extName, err := runtimeclient.ExtensionNameFromHandlerName(*patch.External.DiscoverVariablesExtension)
+				if err != nil {
+					log.Error(err, "failed to reconcile ClusterClass for runtime extensions")
+					continue
+				}
+				if extName == ext.Name {
+					res = append(res, ctrl.Request{NamespacedName: client.ObjectKey{Namespace: clusterClass.Namespace, Name: clusterClass.Name}})
+				}
 				break
```

Comment:
break should go a line up

Reply:
Can you add one CC to the test to cover this? Basically it would be one where only the 2nd referenced discovery handler would match

Reply:
Should be good now - thanks for catching this.

Reply:
You're welcome